src/coreclr/jit/compiler.hpp

```diff
@@ -3190,6 +3190,41 @@ inline bool Compiler::IsValidLclAddr(unsigned lclNum, unsigned offset)
     return (offset < UINT16_MAX) && (offset < lvaLclExactSize(lclNum));
 }
 
+//------------------------------------------------------------------------
+// IsPotentialGCSafePoint: Can the given tree be effectively a gc safe point?
+//
+// Arguments:
+//    tree - the tree to check
+//
+// Return Value:
+//    True if the tree can be a gc safe point
+//
+inline bool Compiler::IsPotentialGCSafePoint(GenTree* tree) const
+{
+    if (((tree->gtFlags & GTF_CALL) != 0))
+    {
+        // if this is not a No-GC helper
+        if (!tree->IsCall() || !emitter::emitNoGChelper(tree->AsCall()->GetHelperNum()))
+        {
+            // assume that we have a safe point.
+            return true;
+        }
+    }
+
+    // TYP_STRUCT-typed stores might be converted into calls (with gc safe points) in Lower.
+    // This is quite a conservative fix as it's hard to prove Lower won't do it at this point.
+    if (tree->OperIsLocalStore())
+    {
+        return lvaTable[tree->AsLclVarCommon()->GetLclNum()].TypeGet() == TYP_STRUCT;
```

Comment:
```suggestion
        return lvaGetDesc(tree->AsLclVarCommon())->TypeGet() == TYP_STRUCT;
```

Feel free to include it in a future change...

Reply:
I think this should check the tree type instead of the local's type, i. e. `return tree->TypeIs(TYP_STRUCT)`. One can have struct stores for primitive locals.

Reply:
Looks simpler so I'm all for it, but it seems unlikely those would end up lowered to calls.

Reply:
Ok, changed